src/components/Contribute/Knowledge/index.tsx

```diff
@@ -90,6 +91,17 @@ export interface KnowledgeFormProps {
 }
 
 export const KnowledgeForm: React.FunctionComponent<KnowledgeFormProps> = ({ knowledgeEditFormData }) => {
+  const [deploymentType, setDeploymentType] = useState<string | undefined>();
+
+  useEffect(() => {
```

Comment:
minor : i think for code readability perspective, it would be good if we move this useEffect() after all the states declarations, where other useEffects are defined? thoughts?

Reply:
Happy either way. I have moved it to follow the convention you suggest.